alacritty_terminal/src/tty/unix.rs

```diff
@@ -121,17 +118,38 @@ impl Pty {
     }
 }
 
-#[cfg(target_os = "macos")]
-fn default_shell(pw: &Passwd<'_>) -> Program {
-    let shell_name = pw.shell.rsplit('/').next().unwrap();
-    let argv = vec![String::from("-c"), format!("exec -a -{} {}", shell_name, pw.shell)];
+fn shell_command(shell_config: Option<&Program>, pw: &Passwd<'_>) -> Command {
+    // Look for a shell in the alacritty config first, then SHELL, then in passwd.
+    let shell = shell_config
+        .map(|prog| prog.program().to_owned())
+        .or_else(|| env::var("SHELL").ok())
+        .unwrap_or_else(|| pw.shell.to_owned());
 
-    Program::WithArgs { program: "/bin/bash".to_owned(), args: argv }
-}
+    #[cfg(not(target_os = "macos"))]
+    let mut command = Command::new(shell);
+
+    // On macOS, unless the user set the `shell` config in alacritty.yml, use the `login` command to
+    // launch the shell so programs like `w`, `who`, and `logname` work.
+    #[cfg(target_os = "macos")]
+    let mut command = if shell_config.is_some() {
+        Command::new(shell)
+    } else {
+        let mut login_command = Command::new("/usr/bin/login");
+        // -f bypasses authentication for the already-logged-in user
+        // -l makes it not a "login" shell and prevents login from changing directories to $HOME
+        // exec the shell with argv[0] prepended by - so the shell run as a login shell anyway
+        login_command.args(["-fl", pw.name, "/bin/zsh", "-c"]);
```

Comment:
I'd say especially for older machines `zsh` is far more likely to not exist than `bash`. Sure bash might be deprecated as a login shell, but I can't imagine they'd get rid of it, breaking any script ever.

I'm not sure if `sh` is just some symlink to one of those, but it certainly would be another option.

Reply:
`/bin/sh` appears to be a minimal GNU `bash`, but not a symlink to `/bin/bash`. I will switch to that rather than pick `zsh`.

I was curious about this though, and it looks like `zsh` has existed in macOS since 10.0:
https://opensource.apple.com/releases/
https://github.com/apple-oss-distributions/distribution-macOS/tree/mac-os-x-100